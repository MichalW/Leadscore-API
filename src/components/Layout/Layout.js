import PropTypes from 'prop-types';
import React from 'react';
import {Container, Grid} from 'semantic-ui-react';

import Navbar from '../Navbar';
import Breadcrumb from '../Breadcrumb';

const Layout = ({children, onSignOut}) => (
  <Container>
    <Navbar onSignOut={onSignOut} />

    <Breadcrumb />

    <Grid centered>
      <Grid.Row>
        <Grid.Column>
          {children}
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Container>
);

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  onSignOut: PropTypes.func.isRequired,
};

export default Layout;
