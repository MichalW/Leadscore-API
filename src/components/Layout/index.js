import {connect} from 'react-redux';

import Layout from './Layout';

const mapStateToProps = state => ({
  signedIn: state.authentication.signedIn,
  username: state.authentication.user.username,
});

export default connect(mapStateToProps)(Layout);
