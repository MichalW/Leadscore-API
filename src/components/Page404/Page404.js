import React from 'react';
import {Segment} from 'semantic-ui-react';

import Layout from '../Layout';

const Page404 = () => (
  <Layout>
    <Segment>
      Page not found (404)
    </Segment>
  </Layout>
);

export default Page404;
