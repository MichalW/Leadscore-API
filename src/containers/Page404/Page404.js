import PropTypes from 'prop-types';
import React from 'react';
import {Segment} from 'semantic-ui-react';

import Header from '../../components/Header';
import Layout from '../../components/Layout';

const Page404 = ({signOutRequest}) => (
  <Layout onSignOut={signOutRequest}>
    <Segment>
      <Header
        icon="times"
        subtitle="Page not found"
        title="404"
      />
    </Segment>
  </Layout>
);

Page404.propTypes = {
  signOutRequest: PropTypes.func.isRequired,
};

export default Page404;
