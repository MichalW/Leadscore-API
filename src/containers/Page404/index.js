import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import Page404 from './Page404';
import {signOutRequest} from '../../actions/authentication';

const mapDispatchToProps = (dispatch) => ({
  signOutRequest: bindActionCreators(signOutRequest, dispatch),
});

export default connect(null, mapDispatchToProps)(Page404);
