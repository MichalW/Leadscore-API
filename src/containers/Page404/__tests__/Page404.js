import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import Page404 from '../Page404';
import Layout from '../../../components/Layout';

it('should render Page404', () => {
  const props = {
    signOutRequest: () => {},
  };

  const page404 = shallow(<Page404 {...props} />);

  expect(page404.find(Layout).exists()).toBe(true);
  expect(toJson(page404)).toMatchSnapshot();
});
