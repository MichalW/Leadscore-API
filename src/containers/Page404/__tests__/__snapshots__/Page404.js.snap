// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Page404 1`] = `
<Layout
  onSignOut={[Function]}
>
  <Segment>
    <HeaderComponent
      icon="times"
      subtitle="Page not found"
      title="404"
    />
  </Segment>
</Layout>
`;
